tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -333,16 +333,44 @@ port::Status GetLoadedCudnnVersion(CudnnVersion* version) {
   return ::tensorflow::OkStatus();
 }
 
-#if CUDNN_MAJOR >= 8 && (CUDNN_MINOR > 0 || CUDNN_PATCHLEVEL >= 4)
-void PreloadCudnnLibrary(cudnnStatus_t (*version_check_fn)(),
-                         absl::string_view sub_library) {
-  cudnnStatus_t status = version_check_fn();
-  if (status != CUDNN_STATUS_SUCCESS) {
-    VLOG(1) << "Could not pre-initialize cuDNN sub-library " << sub_library
-            << ".  Error: " << cudnnGetErrorString(status) << ".";
+// Preload sub libs for cudnn 8.0.4+
+void PreloadCudnnSubLibs(const std::string& kind) {
+#if CUDNN_VERSION >= 8004
+  if (kind == "conv_fwd") {
+    cudnnOpsInferVersionCheck();
+    cudnnCnnInferVersionCheck();
+  } else if (kind == "conv_bwd_filter" || kind == "conv_bwd_data") {
+    cudnnOpsInferVersionCheck();
+    cudnnOpsTrainVersionCheck();
+    cudnnCnnInferVersionCheck();
+    cudnnCnnTrainVersionCheck();
+  } else if (kind == "rnn") {
+    cudnnOpsInferVersionCheck();
+    cudnnOpsTrainVersionCheck();
+    cudnnAdvInferVersionCheck();
+    cudnnAdvTrainVersionCheck();
+  }
+#endif  // CUDNN_VERSION >= 8004
+}
+
+void PreloadCudnnSubLibsHelper(dnn::ConvolutionKind kind) {
+  switch (kind) {
+    case dnn::ConvolutionKind::FORWARD: {
+      PreloadCudnnSubLibs("conv_fwd");
+      break;
+    }
+    case dnn::ConvolutionKind::BACKWARD_DATA: {
+      PreloadCudnnSubLibs("conv_bwd_data");
+      break;
+    }
+    case dnn::ConvolutionKind::BACKWARD_FILTER: {
+      PreloadCudnnSubLibs("conv_bwd_filter");
+      break;
+    }
+    default:
+      break;
```

Comment:
General comment (you decide if relevant here): In general prefer to cover all enum cases individually in a switch. [Guide](https://google.github.io/styleguide/cppguide.html#Loops_and_Switch_Statements) says
> If not conditional on an enumerated value, switch statements should always have a default case (in the case of an enumerated value, the compiler will warn you if any values are not handled). If the default case should never execute, treat this as an error. 

This way should the `ConvolutionKind` ever gets expanded in the future, the compiler will draw the developer's attention here to handle the new case, instead of silently ignoring it.